src/react/index.tsx

```diff
@@ -404,6 +406,29 @@ export function useEmbedRef<T extends EmbedComponent>():
     return React.useRef<React.ComponentRef<T>>(null);
 }
 
+/**
+ *
+ * @param config - EmbedConfig
+ * @returns AuthEventEmitter
+ * @example
+ * ```
+ * function Component() {
+ *  const authEE = useInit({ ...initConfig });
+ *  return <LiveboardEmbed ref={ref} liveboardId={<id>} />
+ * }
+ * ```
+ * @version SDK: 1.36.2 | ThoughtSpot: *
+ */
+export function useInit(config: EmbedConfig) {
+    const ref = useRef<AuthEventEmitter | null>(null);
+    useDeepCompareEffect(() => {
+        const authEE = init(config);
+        ref.current = authEE;
+    }, [config]);
+
+    return ref.current;
```

Comment:
since you are returning the `current` this will always be undefined. You have to return the `ref`